### ISSUE TITLE
训练环境问题

### ISSUE THREAD

**Issue by vampire1204** [zju3dv/EfficientLoFTR#64]
首先非常感谢您的工作！
我在尝试自训练过程中发现一些问题，
     按照readme的 train的env以及requirement 进行安装 似乎会出现各种不兼容的问题？主要涉及pytorch-lightning的版本问题！
     我现在不确定到时是哪的问题
     1)4090卡，torch1.12版本 代码中矩阵运算有问题
     2）3090卡，就出现各种不兼容问题。
     改动部分（对代码中的投影方式以及数据处理方式做了部分改动）
     希望能与解答，谢谢

**Comment by vampire1204** [comment]
Package                                  Version
---------------------------------------- ---------------
absl-py                                  1.4.0
aiofiles                                 23.2.1
aiohttp                                  3.9.1
aiosignal                                1.3.1
albumentations                           0.5.1
anthropic                                0.3.11
anyio                                    3.7.1
appdirs                                  1.4.4
argon2-cffi                              21.3.0
argon2-cffi-bindings                     21.2.0
arrow                                    1.3.0
asgiref                                  3.7.2
astroid                                  3.3.8
asttokens                                2.2.1
async-lru                                2.0.4
async-timeout                            4.0.3
attrs                                    22.2.0
autopep8                                 2.3.1
Babel                                    2.14.0
backcall                                 0.2.0
backoff                                  2.2.1
baron                                    0.10.1
bcrypt                                   4.1.2
beautifulsoup4                           4.11.2
bleach                                   6.0.0
blis                                     0.7.9
boltons                                  23.1.1
bs4                                      0.0.1
build                                    1.0.3
cachetools                               4.2.4
camel-converter                          3.1.1
catalogue                                2.0.8
certifi                                  2022.12.7
cffi                                     1.15.1
chardet                                  2.3.0
charset-normalizer                       3.0.1
chevron                                  0.14.0
chroma-hnswlib                           0.7.3
chromadb                                 0.4.22
click                                    8.1.3
coloredlogs                              15.0.1
comm                                     0.1.2
confection                               0.0.4
continuedev                              0.1.53
contourpy                                1.0.7
cryptography                             41.0.7
cycler                                   0.11.0
cymem                                    2.0.7
Cython                                   0.29.33
debugpy                                  1.6.6
decorator                                5.1.1
defusedxml                               0.7.1
Deprecated                               1.2.14
diffusers                                0.8.0
dill                                     0.3.9
directory-tree                           0.0.3.1
distro                                   1.9.0
docstring-to-markdown                    0.13
einops                                   0.3.0
entrypoints                              0.4
exceptiongroup                           1.2.0
executing                                1.2.0
fastai                                   2.7.11
fastapi                                  0.95.2
fastcore                                 1.5.28
fastdownload                             0.0.7
fastjsonschema                           2.16.3
fastprogress                             1.0.3
filelock                                 3.10.0
fire                                     0.7.0
flatbuffers                              23.5.26
fonttools                                4.39.2
fqdn                                     1.5.1
frozenlist                               1.4.1
fsspec                                   2023.12.2
google-auth                              1.35.0
google-auth-oauthlib                     0.4.6
googleapis-common-protos                 1.62.0
grpcio                                   1.60.0
h11                                      0.14.0
h5py                                     3.1.0
httpcore                                 1.0.2
httptools                                0.6.1
httpx                                    0.26.0
huggingface-hub                          0.16.4
humanfriendly                            10.0
idna                                     3.4
imageio                                  2.26.0
imgaug                                   0.4.0
importlib-metadata                       6.1.0
importlib-resources                      5.12.0
ipykernel                                6.21.3
ipython                                  8.11.0
ipython-genutils                         0.2.0
ipywidgets                               8.0.4
isoduration                              20.11.0
isort                                    5.13.2
jedi                                     0.18.2
Jinja2                                   3.1.3
joblib                                   1.2.0
json5                                    0.9.14
jsonpointer                              2.4
jsonref                                  1.1.0
jsonschema                               4.20.0
jsonschema-specifications                2023.12.1
jupyter                                  1.0.0
jupyter_client                           8.0.3
jupyter-console                          6.6.3
jupyter_core                             5.3.0
jupyter-events                           0.9.0
jupyter-lsp                              2.2.1
jupyter_server                           2.12.4
jupyter_server_terminals                 0.5.1
jupyterlab                               4.0.9
jupyterlab-pygments                      0.2.2
jupyterlab_server                        2.25.2
jupyterlab-widgets                       3.0.5
kiwisolver                               1.4.4
kornia                                   0.4.1
kubernetes                               29.0.0
langcodes                                3.3.0
lazy_loader                              0.1
lightning-lite                           1.8.0
lightning-utilities                      0.11.9
loguru                                   0.5.3
Markdown                                 3.4.1
MarkupSafe                               2.1.3
matplotlib                               3.7.1
matplotlib-inline                        0.1.6
mccabe                                   0.7.0
meilisearch-python-async                 1.8.1
mistune                                  0.8.4
mmh3                                     4.1.0
monotonic                                1.6
mpmath                                   1.3.0
msgpack                                  1.1.0
multidict                                6.0.4
murmurhash                               1.0.9
nbclient                                 0.5.13
nbconvert                                6.4.4
nbformat                                 5.7.3
nest-asyncio                             1.5.6
networkx                                 3.0
notebook                                 6.2.0
notebook_shim                            0.2.3
numpy                                    1.24.2
oauthlib                                 3.2.2
onnxruntime                              1.16.3
openai                                   0.27.10
opencv-python                            4.4.0.46
opentelemetry-api                        1.22.0
opentelemetry-exporter-otlp-proto-common 1.22.0
opentelemetry-exporter-otlp-proto-grpc   1.22.0
opentelemetry-instrumentation            0.43b0
opentelemetry-instrumentation-asgi       0.43b0
opentelemetry-instrumentation-fastapi    0.43b0
opentelemetry-proto                      1.22.0
opentelemetry-sdk                        1.22.0
opentelemetry-semantic-conventions       0.43b0
opentelemetry-util-http                  0.43b0
overrides                                7.4.0
packaging                                23.0
pandas                                   1.5.3
pandocfilters                            1.5.0
parso                                    0.8.3
pathy                                    0.10.1
pexpect                                  4.8.0
pickleshare                              0.7.5
Pillow                                   9.5.0
pip                                      23.3.2
platformdirs                             3.1.1
pluggy                                   1.3.0
posthog                                  3.3.1
preshed                                  3.0.8
prometheus-client                        0.16.0
prompt-toolkit                           3.0.38
protobuf                                 4.22.1
psutil                                   5.9.7
ptyprocess                               0.7.0
pulsar-client                            3.4.0
pure-eval                                0.2.2
pyasn1                                   0.4.8
pyasn1-modules                           0.2.8
pycodestyle                              2.12.1
pycparser                                2.21
pydantic                                 1.10.13
pyDeprecate                              0.3.2
PyGithub                                 1.59.1
Pygments                                 2.14.0
PyJWT                                    2.8.0
pylint                                   3.3.3
PyNaCl                                   1.5.0
pyparsing                                3.0.9
PyPika                                   0.48.9
pyproject_hooks                          1.0.0
pyrsistent                               0.19.3
python-dateutil                          2.8.2
python-dotenv                            1.0.0
python-json-logger                       2.0.7
python-lsp-jsonrpc                       1.1.2
python-lsp-server                        1.9.0
pytorch-lightning                        1.6.0
pytz                                     2022.7.1
PyWavelets                               1.4.1
PyYAML                                   6.0
pyzmq                                    25.0.1
qtconsole                                5.4.1
QtPy                                     2.3.0
ray                                      2.40.0
redbaron                                 0.9.2
referencing                              0.32.1
regex                                    2022.10.31
replicate                                0.11.0
requests                                 2.31.0
requests-oauthlib                        1.3.1
rfc3339-validator                        0.1.4
rfc3986-validator                        0.1.1
ripgrepy                                 2.0.0
rpds-py                                  0.17.1
rply                                     0.7.8
rsa                                      4.9
scikit-image                             0.20.0
scikit-learn                             1.2.2
scipy                                    1.9.1
Send2Trash                               1.8.2
setuptools                               67.4.0
shapely                                  2.0.6
six                                      1.16.0
smart-open                               6.3.0
sniffio                                  1.3.0
socksio                                  1.0.0
soupsieve                                2.4
spacy                                    3.5.1
spacy-legacy                             3.0.12
spacy-loggers                            1.0.4
srsly                                    2.4.6
ssh-import-id                            5.5
stack-data                               0.6.2
starlette                                0.27.0
sympy                                    1.12
tenacity                                 8.2.3
tensorboard                              2.18.0
tensorboard-data-server                  0.7.2
tensorboardX                             1.8
termcolor                                2.5.0
terminado                                0.17.1
testpath                                 0.6.0
thinc                                    8.1.9
threadpoolctl                            3.1.0
tifffile                                 2023.3.15
tiktoken                                 0.4.0
tokenizers                               0.13.2
tomli                                    2.0.1
tomlkit                                  0.13.2
torch                                    1.12.0+cu113
torchaudio                               0.12.0+cu113
torchmetrics                             0.11.4
torchtext                                0.13.0
torchvision                              0.13.0+cu113
tornado                                  6.2
tqdm                                     4.65.0
traitlets                                5.9.0
transformers                             4.25.0
typer                                    0.9.0
types-python-dateutil                    2.8.19.20240106
typing_extensions                        4.5.0
ujson                                    5.9.0
uri-template                             1.3.0
urllib3                                  1.26.15
uvicorn                                  0.21.1
uvloop                                   0.19.0
wasabi                                   1.1.1
watchfiles                               0.21.0
wcwidth                                  0.2.6
webcolors                                1.13
webencodings                             0.5.1
websocket-client                         1.7.0
websockets                               11.0.3
Werkzeug                                 2.1.2
wheel                                    0.38.4
widgetsnbextension                       4.0.5
wrapt                                    1.16.0
xgboost                                  1.7.4
yacs                                     0.1.8
yarl                                     1.9.4
zipp                                     3.15.0

**Comment by vampire1204** [comment]
我使用的是megadepth数据集格式，换了scannet数据集后，环境暂时没有问题
